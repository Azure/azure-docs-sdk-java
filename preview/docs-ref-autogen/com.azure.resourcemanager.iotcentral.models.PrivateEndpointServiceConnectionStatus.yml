### YamlMime:JavaType
uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus"
fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus"
name: "PrivateEndpointServiceConnectionStatus"
nameWithType: "PrivateEndpointServiceConnectionStatus"
summary: "Defines values for Private<wbr>Endpoint<wbr>Service<wbr>Connection<wbr>Status."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.util.ExpandableStringEnum?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedClassMethods:
- classRef: "<xref href=\"com.azure.core.util.ExpandableStringEnum?alt=com.azure.core.util.ExpandableStringEnum&text=ExpandableStringEnum\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.<T>fromString(java.lang.String,java.lang.Class<T>)?alt=com.azure.core.util.ExpandableStringEnum.<T>fromString&text=<T>fromString\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.<T>values(java.lang.Class<T>)?alt=com.azure.core.util.ExpandableStringEnum.<T>values&text=<T>values\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.equals(java.lang.Object)?alt=com.azure.core.util.ExpandableStringEnum.equals&text=equals\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.hashCode()?alt=com.azure.core.util.ExpandableStringEnum.hashCode&text=hashCode\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.toString()?alt=com.azure.core.util.ExpandableStringEnum.toString&text=toString\" data-throw-if-not-resolved=\"False\" />"
- classRef: "<xref href=\"java.lang.Object?alt=java.lang.Object&text=Object\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"java.lang.Object.clone()?alt=java.lang.Object.clone&text=clone\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.finalize()?alt=java.lang.Object.finalize&text=finalize\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.getClass()?alt=java.lang.Object.getClass&text=getClass\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.notify()?alt=java.lang.Object.notify&text=notify\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.notifyAll()?alt=java.lang.Object.notifyAll&text=notifyAll\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.wait()?alt=java.lang.Object.wait&text=wait\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.wait(long)?alt=java.lang.Object.wait&text=wait\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"java.lang.Object.wait(long,int)?alt=java.lang.Object.wait&text=wait\" data-throw-if-not-resolved=\"False\" />"
syntax: "public final class PrivateEndpointServiceConnectionStatus extends ExpandableStringEnum<PrivateEndpointServiceConnectionStatus>"
constructors:
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.PrivateEndpointServiceConnectionStatus()"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.PrivateEndpointServiceConnectionStatus()"
  name: "PrivateEndpointServiceConnectionStatus()"
  nameWithType: "PrivateEndpointServiceConnectionStatus.PrivateEndpointServiceConnectionStatus()"
  syntax: "public PrivateEndpointServiceConnectionStatus()"
fields:
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED"
  name: "APPROVED"
  nameWithType: "PrivateEndpointServiceConnectionStatus.APPROVED"
  summary: "Static value Approved for Private<wbr>Endpoint<wbr>Service<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus?alt=com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus&text=PrivateEndpointServiceConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointServiceConnectionStatus APPROVED"
  desc: "Static value Approved for PrivateEndpointServiceConnectionStatus."
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.PENDING"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.PENDING"
  name: "PENDING"
  nameWithType: "PrivateEndpointServiceConnectionStatus.PENDING"
  summary: "Static value Pending for Private<wbr>Endpoint<wbr>Service<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus?alt=com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus&text=PrivateEndpointServiceConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointServiceConnectionStatus PENDING"
  desc: "Static value Pending for PrivateEndpointServiceConnectionStatus."
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.REJECTED"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.REJECTED"
  name: "REJECTED"
  nameWithType: "PrivateEndpointServiceConnectionStatus.REJECTED"
  summary: "Static value Rejected for Private<wbr>Endpoint<wbr>Service<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus?alt=com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus&text=PrivateEndpointServiceConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointServiceConnectionStatus REJECTED"
  desc: "Static value Rejected for PrivateEndpointServiceConnectionStatus."
methods:
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.fromString(java.lang.String)"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.fromString(String name)"
  name: "fromString(String name)"
  nameWithType: "PrivateEndpointServiceConnectionStatus.fromString(String name)"
  summary: "Creates or finds a Private<wbr>Endpoint<wbr>Service<wbr>Connection<wbr>Status from its string representation."
  modifiers:
  - "static"
  parameters:
  - description: "a name to look for."
    name: "name"
    type: "<xref href=\"java.lang.String?alt=java.lang.String&text=String\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static PrivateEndpointServiceConnectionStatus fromString(String name)"
  desc: "Creates or finds a PrivateEndpointServiceConnectionStatus from its string representation."
  returns:
    description: "the corresponding PrivateEndpointServiceConnectionStatus."
    type: "<xref href=\"com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus?alt=com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus&text=PrivateEndpointServiceConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.values()"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.values()"
  name: "values()"
  nameWithType: "PrivateEndpointServiceConnectionStatus.values()"
  modifiers:
  - "static"
  syntax: "public static Collection<PrivateEndpointServiceConnectionStatus> values()"
  returns:
    description: "known PrivateEndpointServiceConnectionStatus values."
    type: "<xref href=\"java.util.Collection?alt=java.util.Collection&text=Collection\" data-throw-if-not-resolved=\"False\" />&lt;<xref href=\"com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus?alt=com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus&text=PrivateEndpointServiceConnectionStatus\" data-throw-if-not-resolved=\"False\" />&gt;"
type: "class"
desc: "Defines values for PrivateEndpointServiceConnectionStatus."
metadata: {}
package: "com.azure.resourcemanager.iotcentral.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-iotcentral:1.1.0-beta.1
