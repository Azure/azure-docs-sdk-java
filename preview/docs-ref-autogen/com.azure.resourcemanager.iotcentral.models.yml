### YamlMime:JavaPackage
uid: "com.azure.resourcemanager.iotcentral.models"
fullName: "com.azure.resourcemanager.iotcentral.models"
name: "com.azure.resourcemanager.iotcentral.models"
summary: "Package containing the data models for IotCentralClient. Use this API to manage IoT Central Applications in your\n Azure subscription."
classes:
- "com.azure.resourcemanager.iotcentral.models.AppListResult"
- "com.azure.resourcemanager.iotcentral.models.AppPatch"
- "com.azure.resourcemanager.iotcentral.models.AppSku"
- "com.azure.resourcemanager.iotcentral.models.AppSkuInfo"
- "com.azure.resourcemanager.iotcentral.models.AppTemplateLocations"
- "com.azure.resourcemanager.iotcentral.models.AppTemplatesResult"
- "com.azure.resourcemanager.iotcentral.models.CloudErrorBody"
- "com.azure.resourcemanager.iotcentral.models.OperationDisplay"
- "com.azure.resourcemanager.iotcentral.models.OperationInputs"
- "com.azure.resourcemanager.iotcentral.models.OperationListResult"
interfaces:
- "com.azure.resourcemanager.iotcentral.models.App"
- "com.azure.resourcemanager.iotcentral.models.App.Definition"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.Blank"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithDisplayName"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithLocation"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithResourceGroup"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithSku"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithSubdomain"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithTags"
- "com.azure.resourcemanager.iotcentral.models.App.DefinitionStages.WithTemplate"
- "com.azure.resourcemanager.iotcentral.models.App.Update"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages.WithDisplayName"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages.WithSku"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages.WithSubdomain"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages.WithTags"
- "com.azure.resourcemanager.iotcentral.models.App.UpdateStages.WithTemplate"
- "com.azure.resourcemanager.iotcentral.models.AppAvailabilityInfo"
- "com.azure.resourcemanager.iotcentral.models.AppTemplate"
- "com.azure.resourcemanager.iotcentral.models.Apps"
- "com.azure.resourcemanager.iotcentral.models.Operation"
- "com.azure.resourcemanager.iotcentral.models.Operations"
metadata: {}
package: "com.azure.resourcemanager.iotcentral.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-iotcentral:1.0.0-beta.1
