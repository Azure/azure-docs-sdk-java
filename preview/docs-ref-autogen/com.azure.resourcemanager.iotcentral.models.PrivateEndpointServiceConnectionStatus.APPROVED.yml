### YamlMime:JavaMember
uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED*"
fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED"
name: "APPROVED"
nameWithType: "PrivateEndpointServiceConnectionStatus.APPROVED"
members:
- uid: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED"
  fullName: "com.azure.resourcemanager.iotcentral.models.PrivateEndpointServiceConnectionStatus.APPROVED"
  name: "APPROVED"
  nameWithType: "PrivateEndpointServiceConnectionStatus.APPROVED"
  summary: "Static value Approved for PrivateEndpointServiceConnectionStatus."
  syntax: "public static final PrivateEndpointServiceConnectionStatus APPROVED"
type: "field"
metadata: {}
package: "com.azure.resourcemanager.iotcentral.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-iotcentral:1.1.0-beta.1
